src/mongocxx/client_encryption.hpp

```diff
@@ -112,6 +116,115 @@ class MONGOCXX_API client_encryption {
+    ///
+    /// @param id Binary id of which key to delete
+    ///
+    /// @throws mongocxx::exception if there is an error getting the key.
+    ///
+    /// @return The result of the internal find() operation on the key vault collection.
+    ///
+    /// @see https://www.mongodb.com/docs/manual/reference/method/KeyVault.getKey/
+    ///
+    stdx::optional<bsoncxx::document::value> get_key(bsoncxx::types::bson_value::view_or_value id);
+
+    ///
+    /// Finds all documents in the key vault collection.
+    ///
+    /// @throws mongocxx::exception if there is an error getting the keys.
+    ///
+    /// @return the result of the internal find() operation on the key vault collection.
+    ///
+    /// @see https://www.mongodb.com/docs/manual/reference/method/KeyVault.getKeys/
+    ///
+    mongocxx::cursor get_keys();
+
+    ///
+    /// Adds a keyAltName to the keyAltNames array of the key document in the
+    /// key vault collection with the given UUID (BSON binary subtype 0x04).
+    ///
+    /// @param id Binary id of the key to add the key alternate name to
+    ///
+    /// @param key_alt_name String alternative name for the key
+    ///
+    /// @throws mongocxx::exception if there is an error adding the key alt name.
+    ///
+    /// @return the previous version of the key document.
+    ///
+    /// @see https://www.mongodb.com/docs/manual/reference/method/KeyVault.addKeyAlternateName/
+    ///
+    stdx::optional<bsoncxx::document::value> add_key_alt_name(
+        bsoncxx::types::bson_value::view_or_value id, bsoncxx::string::view_or_value key_alt_name);
+
+    ///
+    /// Removes a keyAltName from the keyAltNames array of the key document in
+    /// the key vault collection with the given UUID (BSON binary subtype 0x04).
+    ///
+    /// @param id Binary id of the key to remove the key alternate name from
+    ///
+    /// @param key_alt_name String alternative name for the key
+    ///
+    /// @throws mongocxx::exception if there is an error removing the key alt name.
+    ///
+    /// @return The previous version of the key document.
+    ///
+    /// @see https://www.mongodb.com/docs/manual/reference/method/KeyVault.removeKeyAlternateName/
+    ///
+    stdx::optional<bsoncxx::document::value> remove_key_alt_name(
+        bsoncxx::types::bson_value::view_or_value id, const std::string& key_alt_name);
+
+    ///
+    /// Get the key document from the key vault collection with the provided name.
+    ///
+    /// @param key_alt_name String alternative name for the key
+    ///
+    /// @throws mongocxx::exception if there is an error getting the key by alt name.
+    ///
+    /// @return A key document in the key vault collection with the given keyAltName.
+    ///
+    /// @see https://www.mongodb.com/docs/manual/reference/method/KeyVault.getKeyByAltName/
+    ///
+    stdx::optional<bsoncxx::document::value> get_key_by_alt_name(const std::string& key_alt_name);
```

Comment:
Use `bsoncxx::string::view_or_value` in _all_ new public API functions for consistency.